pinot-core/src/main/java/org/apache/pinot/core/operator/combine/GroupByOrderByCombineOperator.java

```diff
@@ -197,6 +205,13 @@ protected void processSegments(int taskIndex) {
     }
   }
 
+  // Check for thread interruption, every time after merging 10_000 keys
+  private void checkMergePhaseInterruption(long mergedKeys) {
+    if (mergedKeys % DocIdSetPlanNode.MAX_DOC_PER_CALL == 0 && Thread.interrupted()) {
```

Comment:
(minor) Let's use a separate constant for this

Reply:
fixed